File: 백준/Silver/1920. 수 찾기/수 찾기.py
def binary_search(array, target, start, end):
    while start <= end:
        mid = (start + end) // 2

        if array[mid] == target:
            return mid
        elif array[mid] > target:
            end = mid - 1
        else:
            start = mid + 1
    return None

N = int(input())

A = list()

A = list(map(int, input().split()))

A.sort()

M = int(input())

B = list(map(int, input().split()))

for i in range(M):
    answer = binary_search(A, B[i], 0, N - 1)
    if answer == None:
        print(0)
    else:
        print(1)

File: 백준/Silver/1920. 수 찾기/수 찾기.cs
using System;
using System.Text;

namespace Baekjoon {
    class Program {
        static void Main()
        {
            int n = int.Parse(Console.ReadLine());
            List<int> nums1 = Console.ReadLine().Split().Select(int.Parse).ToList<int>();
            
            int m = int.Parse(Console.ReadLine());
            List<int> nums2 = Console.ReadLine().Split().Select(int.Parse).ToList<int>();

            nums1.Sort();

            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < m; i++)
            {
                sb.AppendLine(IsInclude(n, nums1, nums2[i]).ToString());
            }

            Console.WriteLine(sb.ToString());
        }

        static int IsInclude(int size, List<int> nums, int target)
        {
            int start = 0;
            int end = size-1;
            
            while (start <= end)
            {
                int mid = (end + start) / 2;
                
                if (nums[mid] == target)
                {
                    return 1;
                }
                
                if (nums[mid] > target)
                {
                    end = mid - 1;
                }
                else if (nums[mid] < target)
                {
                    start = mid + 1;
                }
            }

            return 0;
        }
    }
}